src/charm.py

```diff
@@ -655,7 +661,9 @@ def _define_ingresses(self, excluded_relation=None) -> None:  # type: ignore[no-
         # Filter out any cases in which we don't have data set (e.g.: missing relation data)
         config_or_relations = filter(self._has_required_fields, config_or_relations)
 
-        ingresses = [conf_or_rel._get_k8s_ingress() for conf_or_rel in config_or_relations]
+        ingresses = [
+            conf_or_rel._get_k8s_ingress(self.app.name) for conf_or_rel in config_or_relations
+        ]
 
         ingresses = self._process_ingresses(ingresses)
```

Comment:
What processing does this function do? Maybe the result should be assigned to a different variable name since what it points to has changed to now be processed?

Reply:
I don't follow, `_process_ingresses` doesn't do anything?

Reply:
It creates the actual Ingress objects from the data defined in define_ingresses.